xapian-applications/omega/docs/omegascript.rst

```diff
@@ -222,10 +222,10 @@ $fmt
 $freq{term}
 	frequency of a term
 
-$hash{TEXT[,HASH}
+$hash{TEXT[,HASH]}
     unique ID for ``TEXT`` string using the hashing algorithm specified by
-    ``HASH`` which defaults to MD5 hashing algorithm. Currently, this command
-    only supports MD5 which yields a 128-bit hash sequence of 32 hexadecimal
+    ``HASH`` which defaults to MD5. Currently, this command only supports
+    MD5 which yields a 128-bit hash sequence serialised as 32 hexadecimal
     characters.
```

Comment:
By having a default and making that `md5` I'm slightly concerned that we actually tie ourselves to supporting MD5 for longer, because changing the default causes "quiet" problems.  If we drop support for an algorithm, at least that should make templates which specify that algorithm explicitly fail with an error.  It seems likely we'll eventually end up in a situation where basically everyone has to specify the hash to use (because even now, for most uses you don't want MD5 and that's not going to become less true with time), but we can't change the default because that could break templates for someone not specifying the hash but assuming the default is MD5 (not an unreasonable assumption, as we're documenting this above).

So I'm not entirely sure if having a default for HASH is useful or not - it seems to really depend if there are cases where you want to hash something but you really don't care what hash algorithm is used, at least to the extent that you might use a different algorithm tomorrow.  What use cases did whoever suggested having a default have in mind?

Also, although the docs say HASH can be omitted, the command table in `query.cc` still marks `$hash` as taking at least 2 parameters.  If we are going to support a default, then that needs fixing and there should be a testcase to check that the defaulting actually works.